src/diffusers/models/attention_processor.py

```diff
@@ -593,7 +597,7 @@ def get_attention_scores(
             beta = 0
         else:
             baddbmm_input = attention_mask
-            beta = 1
+            beta = self.scale_mask_factor
```

Comment:
added a new config for `Attention` here

set this to be a large negative number helps a lot with numerical stability. in kandinsky they "fill" the empty tokens 
in attention_matrix with largest possible negative number(see code -> 
https://github.com/huggingface/diffusers/blob/5ae3c3a56b52094cd01d184da38255136d7715e7/src/diffusers/models/attention_processor.py#L2251)

I set this config to be `-6000` for simplicity - not exactly same but seem to be sufficient.

Reply:
Hmm is `beta` supposed to be used to control mask precision?

Reply:
actually, I think I should do this instead! https://github.com/huggingface/diffusers/blob/d1b2a1a95710964ff84b2c2332cd2ad93734561e/src/diffusers/models/unet_2d_condition.py#L896

This `beta` here is essentially trying to do the same thing - it's trying to zero out the zero token's attention score in the softmax operation. I did not realize I was missing this step because Kandinsky cuts off (most except for one) the zero tokens from `prompt_embeds`, so not doing this step still generates accurate output for the most part

Reply:
refactored and now this script (one scenario when the attention_mask actually need to be applied) not exactly but similar outputs on main and branch 

```python
from diffusers import AutoPipelineForText2Image
import torch

pipe = AutoPipelineForText2Image.from_pretrained("kandinsky-community/kandinsky-3", variant="fp16", torch_dtype=torch.float16)
pipe.enable_model_cpu_offload()
        
prompt = ["A photograph of the inside of a subway train. There are raccoons sitting on the seats. One of them is reading a newspaper. The window shows the city in the background.",
          "A photograph of the inside of a subway train. There are raccoons sitting on the seats. One of them is reading a newspaper. The window shows the city in the background. A photograph of the inside of a subway train. There are raccoons sitting on the seats. One of them is reading a newspaper. The window shows the city in the background."]

generator = [torch.Generator(device="cpu").manual_seed(0),torch.Generator(device="cpu").manual_seed(1)]
image = pipe(prompt, num_inference_steps=25, generator=generator).images[0]
```

main         |  branch
:-------------------------:|:-------------------------:
![yiyi_test_3_out_bm](https://github.com/huggingface/diffusers/assets/12631849/e04f0091-ce73-43c0-8153-8c5db088e473) | ![yiyi_test_3_out](https://github.com/huggingface/diffusers/assets/12631849/34d03207-63cb-4dba-aeaf-cdc46d8a8d5c)